### DISCUSSION TITLE
Conversion to world coordinates

### DISCUSSION THREAD

**Discussion by kakeru58** [Victorlouisdg/cloth-competition#21] [Questions] [answered]
How can I convert x,y,z coordinates acquired with an RGB-D camera into a world coordinate system?

**Comment by Victorlouisdg** [comment, marked as answer]
Hi Kakeru

I assume you have already used the depth to map to "unproject" pixel coordinates to points in 3D. However, these 3D points are expressed in the camera's coordinate frame. To get the coordinates of those 3D points expressed in the world frame used for the competition, you need to apply a transformation:
```
point_in_world = camera_pose_in_world @ point_in_camera
```
Where:
* `camera_pose_in_world` a 4x4 homogeneous matrix representing the camera's pose expressed in the world frame. This is included in the `CompetitionObservation` dataclass and can be used like this: `observation.camera_pose_in_world`
* `point_in_camera` is a 4x1 homogeneous point, which I assume is the point you already have calculated
* `point_in_world` is a 4x1 homogeneous point, the result, which is the point expressed in the world frame.

Alternatively, you can also have a look at how I do the same thing in the [`calculate_grasp_pose_from_annotations()`](https://github.com/Victorlouisdg/cloth-competition/blob/c89426b94235c0371a2c9c14326a1918231ee329/cloth-tools/cloth_tools/annotation/grasp_annotation.py#L155) function.

**Reply by kakeru58** [reply]
Hi! Victor-Louis De Gusseme
Thank you so much!
I got it!
